### ISSUE TITLE
@argument directive List types not loaded.

### ISSUE THREAD

**Issue by sjcobb2022** [HoudiniGraphql/houdini#920]
### Describe the bug

I am currently trying use `@arguments` on a large database of products. There are a series of filters that are passed into a query as an array. (e.g. `[ProductFilters]`). For versatility, I currently have these inputs in a fragment, which uses the `@arguments` and `@with` directives.

However, I get the error: 

`❌ Could not unwrap type: {"kind":"NamedType","name":{"kind":"Name","value":"[ProductFilter]"}}`

After looking through the code it looks like there may be an error in the lib/graphql.ts file 

https://github.com/HoudiniGraphql/houdini/blob/a8237aa5c309de68b126ed55c6fe3fd6f1b24503/packages/houdini/src/lib/graphql.ts#L203-L237

I believe that there is some sort of logical error with the parsing of the array.

I would also like to point out another bug which is that if I remove the quotes from the array-type, no error is thrown, however types are not generated for the the input. 

```graphql
@arguments(
	productsFirst: { type: "Int", default: 5 }
	productsAfter: { type: "String" }
	productsFilters: { type: [ProductFilter] } # This doesn't throw an error. Should throw "type field to @arguments must be a string"
)
```

### Severity

serious, but I can work around it

### Steps to Reproduce the Bug

1. Create a graphql fragment that takes an input of an array
2. use `@arguments` with a type of the input e.g `@arguments(test: {type: "[Int]"})`
3. Get error  with passing.

### Reproduction

Can't find a good gql endpoint that has inputs of Lists as a query param.

_No response_

## ALSO CONGRATS ON 1.0!

**Comment by AlecAivazis** [comment]
Thanks!! 😄 

I'll find some time in the next few days to look into this. As for the lack of validation, its one of the oldest remaining issues (#120). Will likely try to knock out both of these together 👍

**Comment by AlecAivazis** [comment]
Both of these should be fixed in `v1.0.1`. Thanks for reporting it!

**Comment by sjcobb2022** [comment]
> Both of these should be fixed in `v1.0.1`. Thanks for reporting it!

Wow that was fast!

Thank you so much!
